src/GitHub.App/Services/ModelService.cs

```diff
@@ -157,22 +157,33 @@ public IObservable<AccountCacheItem> GetUserFromCache()
             var keyobs = GetUserFromCache()
                 .Select(user => string.Format(CultureInfo.InvariantCulture, "{0}|{1}:{2}", CacheIndex.PRPrefix, user.Login, repo.Name));
 
-            var source = Observable.Defer(() => keyobs
-                .SelectMany(key =>
-                    hostCache.GetAndFetchLatestFromIndex(key, () =>
+            var source = Observable.Defer(() =>
+            {
+                var collectionIsDisposed = false;
+
+                return keyobs
+                    .SelectMany(key => hostCache.GetAndFetchLatestFromIndex(key, () =>
                         apiClient.GetPullRequestsForRepository(repo.CloneUrl.Owner, repo.CloneUrl.RepositoryName)
-                                 .Select(PullRequestCacheItem.Create),
-                        item =>
-                        {
-                            // this could blow up due to the collection being disposed somewhere else
-                            try { collection.RemoveItem(Create(item)); }
-                            catch (ObjectDisposedException) { }
-                        },
-                        TimeSpan.Zero,
-                        TimeSpan.FromDays(7))
-                )
-                .Select(Create)
-            );
+                            .Select(PullRequestCacheItem.Create), item =>
+                            {
+                                try
```

Comment:
I am totally fine with adding a `public bool Disposed => disposed;` property in `TrackingCollection` so we can check here if it's disposed instead of waiting for the exception. We still need the try catch block (because race conditions), but in the majority of cases we should be able to avoid exceptions being thrown completely, which would be nice.